### ISSUE TITLE
SyntaxError: Unexpected strict mode reserved word

### ISSUE THREAD

**Issue by simon-p-r** [hapijs/lab#465]
Hi

I am encountering above error when testing this pseudo code.

```JS

var store = new Map();
store.set('key', value)
 for (let item of store) {
     // code here
 }
````

I had a look at espree config but couldn't find anything that would cause this.

**Comment by geek** [comment]
@simon-p-r what other details can you provide: node version, how you are running lab, and what does the full test look like.

**Comment by simon-p-r** [comment]
@geek as requested

````js
node version 4.1
lab -a code -v -t 100 -L
````

I am not testing function directly at the moment but function code is in looks like this

````js

internals.parseSchema = function (obj, match, lookup) {

    if (!internals.isObj(obj)) {
        return;
    }
    var store = new Map();
    var newKey;
    var keys = Object.keys(obj);
    for (var i = 0, il = keys.length; i < il; ++i) {

        var key = keys[i];
        var node = obj[key];
        store.set(key, node);

        if (internals.isObj(node)) {

            internals.parseSchema(node, match, lookup);
        }
        if (internals.isString(node)) {

            for (let item of store) {
                if (item[0].indexOf(match) !== -1) {

                    delete obj[key];
                    newKey = item[0].split('.').slice(-1).join('');
                    obj[newKey] = lookup[item[1]].schema;
                }
            }
        }
        if (internals.isArray(node)) {

            for (var a = 0, al = node.length; a < al; ++a) {

                var subnode = node[a];
                store.set(key, subnode);

                if (internals.isObj(subnode)) {
                    internals.parseSchema(subnode, match, lookup);
                }
                if (internals.isString(subnode)) {

                    if (subnode.indexOf(match) !== -1) {

                        newKey = subnode.split('.').slice(-1).join('');
                        obj[key][a] = lookup[newKey].schema;

                    }
                }

            }
        }
    }
    return obj;

};

````

**Comment by cjihrig** [comment]
That error generally happens when you use a strict mode reserved word as a variable name. For example:

```javascript
'use strict';
var yield = 5;
```

Isn't there a stack trace or anything that tells you where the problem is?

**Comment by simon-p-r** [comment]
Here is stack trace

````js
D:\Modules\new-yajsv>npm t

> new-yajsv@0.0.1 test D:\Modules\new-yajsv
> lab -a code -v -t 95 -L

D:\Modules\new-yajsv\lib\parse.js:29
            for (global.__$$labCov._line('D:/Modules/new-yajsv/lib/parse.js',29);let item of store) {
                                                                                 ^^^

SyntaxError: Unexpected strict mode reserved word
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:414:25)
    at Object.require.extensions.(anonymous function) [as .js] (D:\Modules\new-yajsv\node_modules\lab\lib\coverage.js:29:36)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (D:\Modules\new-yajsv\lib\index.js:4:76)
    at Module._compile (module.js:435:26)
    at Object.require.extensions.(anonymous function) [as .js] (D:\Modules\new-yajsv\node_modules\lab\lib\coverage.js:29:36)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (D:\Modules\new-yajsv\test\index.js:5:15)
    at Module._compile (module.js:435:26)
    at Object.require.extensions.(anonymous function) [as .js] (D:\Modules\new-yajsv\node_modules\lab\lib\coverage.js:34:28)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at D:\Modules\new-yajsv\node_modules\lab\lib\cli.js:104:23
    at Array.forEach (native)
    at Object.internals.traverse (D:\Modules\new-yajsv\node_modules\lab\lib\cli.js:100:19)
    at Object.exports.run (D:\Modules\new-yajsv\node_modules\lab\lib\cli.js:49:29)
    at Object.<anonymous> (D:\Modules\new-yajsv\node_modules\lab\bin\lab:3:23)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:134:18)
    at node.js:961:3
npm ERR! Test failed.  See above for more details.
````
I am not using map and let of item to work around issue.

**Comment by cjihrig** [comment]
Does the following code work for you in Node without lab:

```javascript
'use strict';
var store = new Map();

for (let foo of store) {
    console.log(foo);
}
```

It should. And can you make sure you're on the latest version of lab (6.x.x). This works for me with latest lab on OS X.

**Comment by simon-p-r** [comment]
Yes this works on my Windows box, the code above that causes issues works outside of lab with node.

**Comment by Marsup** [comment]
Lab is changing the structure of the for..of, I guess it didn't understand the let keyword as well as it understood var.

**Comment by simon-p-r** [comment]
Sorry I didn't say it also occurs if I use var instead of let

**Comment by simon-p-r** [comment]
Fixed by cb999c1376b88b4fc22105abc0d226b2adda759f
